internal/backends/python/python.go

```diff
@@ -311,6 +304,167 @@ func makePythonPoetryBackend(python string) api.LanguageBackend {
 	}
 }
 
+var pythonGuessRegexps = util.Regexps([]string{
+	// The (?:.|\\\n) subexpression allows us to
+	// match match multiple lines if
+	// backslash-escapes are used on the newlines.
+	`from (?:.|\\\n) import`,
+	`import ((?:.|\\\n)*) as`,
+	`import ((?:.|\\\n)*)`,
+})
+
+// makePythonPipBackend returns a backend for invoking poetry, given an arg0 for invoking Python
+// (either a full path or just a name like "python3") to use when invoking Python.
+func makePythonPipBackend(python string) api.LanguageBackend {
+	var pipFlags []PipFlag
+
+	b := api.LanguageBackend{
+		Name:                 "python3-pip",
+		Specfile:             "requirements.txt",
+		Alias:                "python-python3-pip",
+		FilenamePatterns:     []string{"*.py"},
+		Quirks:               api.QuirksAddRemoveAlsoInstalls | api.QuirksNotReproducible,
+		NormalizePackageName: normalizePackageName,
+		GetPackageDir: func() string {
+			// Check if we're already inside an activated
+			// virtualenv. If so, just use it.
+			if venv := os.Getenv("VIRTUAL_ENV"); venv != "" {
+				return venv
+			}
+
+			if outputB, err := util.GetCmdOutputFallible([]string{
+				"python",
+				"-c", "import site; print(site.USER_SITE)",
+			}); err == nil {
+				return string(outputB)
+			}
```

Comment:
should we check if a venv is available before falling back to user site?

Reply:
I had that initially, but removed it in favor of https://github.com/replit/upm/pull/180#discussion_r1425890670

I agree with this, actually, since if the user `activate`'s the venv before running `upm add`, they'll install into the venv, if they don't they won't. Compare that to only _one_ subcommand activating the venv, giving inconsistent behavior.

Reply:
good point!